consensus/safety-rules/src/tests/suite.rs

```diff
@@ -958,3 +964,51 @@ fn test_2chain_timeout(constructor: &Callback) {
         Error::NotSafeToTimeout(4, 1, 3, 2)
     );
 }
+
+/// Test that we can succesfully sign a valid commit vote
+fn test_sign_commit_vote(constructor: &Callback) {
+    // we construct a chain of proposals
+    // genesis -- a1 -- a2 -- a3
+
+    let (mut safety_rules, signer, key) = constructor();
+    let (proof, genesis_qc) = test_utils::make_genesis(&signer);
+
+    let round = genesis_qc.certified_block().round();
+    safety_rules.initialize(&proof).unwrap();
+
+    let a1 = test_utils::make_proposal_with_qc(round + 1, genesis_qc, &signer, key.as_ref());
+    let a2 = make_proposal_with_parent(round + 2, &a1, None, &signer, key.as_ref());
+
+    // now we try to agree on genesis's execution result
```

Comment:
I think genesis is similar to BlockInfo::empty and should fail the test, we should add a test case for non-genesis block info

Reply:
Resolved